CovertInfiltration/Src/CovertInfiltration/Classes/UICovertActionsGeoscape.uc

```diff
@@ -0,0 +1,964 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  Xymanek
+//  PURPOSE: This is a new (custom) screen for covert ops that uses the world map
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class UICovertActionsGeoscape extends UIScreen;
+
+// UI - list
+var UIList ActionsList;
+
+// UI - layout. Note that there is no need for left pane since UIList is a container on its own
+var UIPanel CenterSection;
+var UIMask CenterSectionMask; // Used to animate in
+var UIPanel RightPane;
+
+// UI - action info (top)
+var UIPanel ActionInfoTopContainer;
+var UIBGBox ActionInfoBG;
+var UIImage ActionDisplayNameBG;
+var UIText ActionDisplayName;
+var UIText ActionDescription;
+
+// UI - action info (bottom)
+var UIPanel ActionInfoBottomContainer;
+
+// UI - action reward
+var UIPanel ActionRewardContainer;
+var UIImage ActionRewardHeaderBG;
+var UIText ActionRewardHeader;
+var UIPanel ActionRewardTextBG;
+var UIText ActionRewardText;
+
+// UI - action slots
+var UIPanel ActionSlotsContainer;
+var UIImage ActionSlotsHeaderBG;
+var UIText ActionSlotsHeader;
+var UIPanel ActionSlotsTextBG;
+var UIList ActionSlotRows;
+
+// UI - action image
+var UIImage ActionImageBorder;
+var UIImage ActionImage;
+
+// UI - buttons
+var UIPanel ButtonGroupWrap;
+var UIBGBox ButtonsBG;
+var UIButton ConfirmButton, CloseScreenButton;
+
+// UI - action risks
+var UIPanel ActionRisksContainer;
+var UIImage ActionRisksHeaderBG;
+var UIText ActionRisksHeader;
+var UIPanel ActionRisksTextBG;
+var UIText ActionRisksText;
+
+// Data
+var StateObjectReference ActionRef;
+var array<XComGameState_CovertAction> arrActions;
+var StateObjectReference ActionToShowOnInitRef;
+
+// SquadSelect manager
+var protectedwrite UISSManager_CovertAction SSManager;
+
+// Set by UISS controller
+var bool bConfirmScreenWasOpened;
+
+// Pre-open values
+var protected bool bPreOpenResNetForcedOn;
+var protected EStrategyMapState PreOpenMapState;
+
+// Internal state
+var protected bool bDontUpdateData;
+
+const ANIMATE_IN_DURATION = 0.5f;
+const CAMERA_ZOOM = 0.5f;
+
+const UI_INFO_BOX_MARGIN = 10;
+const ACTION_SLOTS_PER_ROW = 2;
+
+///////////////////////
+/// Creating screen ///
+///////////////////////
+
+simulated function InitScreen(XComPlayerController InitController, UIMovie InitMovie, optional name InitName)
+{
+	super.InitScreen(InitController, InitMovie, InitName);
+
+	// Testing code
+	if (ActionToShowOnInitRef.ObjectID != 0) bAnimateOnInit = false;
+
+	GetHQPres().StrategyMap2D.Hide();
+	GetHQPres().CAMSaveCurrentLocation();
+	OnInitForceResistanceNetwork();
+
+	BuildScreen();
+}
+
+simulated function OnInit()
+{
+	super.OnInit();
+
+	// We can have 0-n list index changes during bootsratp so we call UpdateData manually once (!) after we are done
+	bDontUpdateData = true;
```

Comment:
This looks ugly. Is there a better way? Maybe you should use `OnSelectionChanged` instead of  `OnSetSelectedIndex` for the UIList?

Reply:
Not super happy about this solution either, but it works... We can fix it in another PR, I want to get this massive PR out of the way